tracer/src/Datadog.Trace/Span.cs

```diff
@@ -405,17 +405,25 @@ internal void SetExceptionTags(Exception exception)
         {
             if (exception != null)
             {
-                // for AggregateException, use the first inner exception until we can support multiple errors.
-                // there will be only one error in most cases, and even if there are more and we lose
-                // the other ones, it's still better than the generic "one or more errors occurred" message.
-                if (exception is AggregateException aggregateException && aggregateException.InnerExceptions.Count > 0)
+                try
                 {
-                    exception = aggregateException.InnerExceptions[0];
-                }
+                    // for AggregateException, use the first inner exception until we can support multiple errors.
+                    // there will be only one error in most cases, and even if there are more and we lose
+                    // the other ones, it's still better than the generic "one or more errors occurred" message.
+                    if (exception is AggregateException aggregateException && aggregateException.InnerExceptions.Count > 0)
+                    {
+                        exception = aggregateException.InnerExceptions[0];
+                    }
 
-                SetTag(Trace.Tags.ErrorMsg, exception.Message);
-                SetTag(Trace.Tags.ErrorStack, exception.ToString());
-                SetTag(Trace.Tags.ErrorType, exception.GetType().ToString());
+                    SetTag(Trace.Tags.ErrorMsg, exception.Message);
```

Comment:
Small nit: It might be easier to move the exception null-check until after the AggregateException if statement, since that if statement _also_ functions as a null-check

Reply:
Does it make any difference 🤔 The current order is what we had previously, and I think you always need _both_ checks, but with the current order, if it's null you can bail out straight away. With the other order, you would always do 2 checks I think?

Reply:
Yeah I suppose you're right, this is fine as-is 👍🏼